invenio_circulation/transitions/base.py

```diff
@@ -30,22 +29,22 @@ def inner(self, loan, **kwargs):
         new_item_pid = kwargs.get('item_pid')
 
         if not current_app.config['CIRCULATION_ITEM_EXISTS'](new_item_pid):
-            msg = 'Item `{0}` not found in the system'.format(new_item_pid)
-            raise TransitionConstraintsViolation(msg=msg)
+            msg = "Item '{0}' not found in the system".format(new_item_pid)
+            raise ItemNotAvailable(description=msg)
 
         if loan.get('item_pid') and new_item_pid != loan['item_pid']:
-            msg = 'Loan item is `{0}` but transition is trying to set it to ' \
-                  '`{1}`'.format(loan['item_pid'], new_item_pid)
-            raise TransitionConstraintsViolation(msg=msg)
+            msg = "Loan item is '{0}' but transition is trying to set it to " \
+                  "'{1}'".format(loan['item_pid'], new_item_pid)
+            raise TransitionConstraintsViolation(description=msg)
```

Comment:
I think this should be set to a new exception as by raising `TransitionConstraintsViolation` it will trigger the check of the next available transition and probably we need to exit here? WDYT @ntarocco @kprzerwa ?